starknet_devnet/compile.py

```diff
@@ -0,0 +1,50 @@
+"""Compilation utilities"""
+
+import json
+import os
+import subprocess
+import tempfile
+
+from starkware.starknet.definitions.error_codes import StarknetErrorCode
+from starkware.starknet.services.api.contract_class.contract_class import (
+    CompiledClass,
+    ContractClass,
+)
+
+from starknet_devnet.contract_class_utils import load_casm
+from starknet_devnet.util import StarknetDevnetException
+
+
+def compile_cairo(
+    contract_class: ContractClass, compiler_manifest: str
+) -> CompiledClass:
+    """Compile ContractClass to CompiledClass using compiler_manifest"""
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        contract_json = os.path.join(tmp_dir, "contract.json")
+        contract_casm = os.path.join(tmp_dir, "contract.casm")
+
+        with open(contract_json, mode="w", encoding="utf-8") as tmp_file:
+            contract_class_dumped = contract_class.dump()
+            contract_class_dumped["abi"] = json.loads(contract_class_dumped["abi"])
+            json.dump(contract_class_dumped, tmp_file)
+
+        compilation_args = [
```

Comment:
Is there anyway we can add `--allowed-libfuncs-list-name experimental_v0.1.0` here? We seem to get errors otherwise.

Reply:
Thanks for following this PR and testing Devnet in experimental mode! I hardcoded the passing of this flag with commit https://github.com/Shard-Labs/starknet-devnet/pull/415/commits/c63aabd4fdd517f1d2cce61ea079c22e4a68e5ff.

It might be more flexible to provide another CLI option where the exact list of libfuncs would be specified. Although that induce more CLI bloating